src/compiler/checker.ts

```diff
@@ -8255,6 +8269,10 @@ namespace ts {
             return type && (type.flags & TypeFlags.Any) !== 0;
         }
 
+        function isErrorType(type: Type) {
+            return type === errorType || !!(type.flags & TypeFlags.Any && type.aliasSymbol);

```

Comment:
No, that wouldn't work. I'll add a clarifying comment, but basically we want to identify the special `TypeFlags.Any` types produced by getTypeForTypeAliasReference for an unresolved symbol. The sole purpose of `unresolvedType` is to act as the declared type for the special type alias symbols we create for unresolved names. The type is never actually returned by anything because getTypeForTypeAliasReference maps it into a special `TypeFlags.Any` type with an `aliasSymbol`.